catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/DynamicSchemaResolverTest.java

```diff
@@ -92,6 +97,43 @@ public void testAddFields() throws Exception {
     }
   }
 
+  /**
+   * Verify that even unchecked exceptions generating xpath support do not prevent a metacard from
+   * being stored.
+   */
+  @Test
+  public void testBufferOverflow() throws MetacardCreationException {
+    // Setup
+    String metacardTypeName = "states";
+    Set<AttributeDescriptor> attributeDescriptors = new HashSet<>(1);
+    String attributeName = Metacard.METADATA;
+    attributeDescriptors.add(new CoreAttributes().getAttributeDescriptor(attributeName));
+    //    char[] metadata = new char[100000000];
+    //    Arrays.fill(metadata, 'a');
+    StringBuilder mockValue = new StringBuilder();
+    mockValue.append("<?xml version=\"1.1\" encoding=\"UTF-32\"?><metadata></metadata>");
+    Attribute mockAttribute = new AttributeImpl(Metacard.METADATA, mockValue.toString());
+    Metacard mockMetacard = mock(Metacard.class, RETURNS_DEEP_STUBS);
+    when(mockMetacard.getMetacardType().getName()).thenReturn(metacardTypeName);
+    when(mockMetacard.getMetacardType().getAttributeDescriptors()).thenReturn(attributeDescriptors);
+    when(mockMetacard.getAttribute(attributeName)).thenReturn(mockAttribute);
+    when(mockMetacard.getMetadata()).thenReturn(mockValue.toString());
+    SolrInputDocument mockSolrInputDocument = mock(SolrInputDocument.class);
+    DynamicSchemaResolver resolver =
+        new DynamicSchemaResolver() {
+          @Override
+          protected byte[] createTinyBinary(String xml)
+              throws XMLStreamException, SaxonApiException, IOException {
+            throw new BufferOverflowException();
+          }
+        };
+
+    // Perform Test
+    resolver.addFields(mockMetacard, mockSolrInputDocument);
+
```

Comment:
✏️ Maybe also verify that your mock metacard's `getId()` method is invoked once?

Reply:
❓ Should we also verify that `solrInputDocument.addField(LUX_XML_FIELD_NAME, luxXml);` wasn't called?

Reply:
Technically we should also verify that all the other attributes were properly added and nothing is missing. Which I think in this case is only `METACARD_TYPE_OBJECT_FIELD_NAME`.

Reply:
Do we really want to verify getId is called?? If someone else comes along and modifies `addFields` that uses the ID for something else then wouldn't this test fail? Also it seems like checking for that getId is just a roundabout way of checking if the logging side effect happened :/

Reply:
Since we don't have a way to tell an exception that we log and otherwise ignore has occurred, it seems our best bet here. I've seen too many unit tests that mistakenly assert behaviors have occurred when they haven't. I don't love the verification, but I can't think of anything that's cleaner.

Reply:
Since we changed the behavior to silence the internal exception, I'd think the test should just validate it is not throwing an exception plus the lux field wasn't added.

Reply:
I definitely agree with the latter, but the exception also wouldn't be thrown under normal operation (the subclassed `DynamicSchemaResolver` notwithstanding). I suppose checking that we didn't add the lux field would be sufficient to assert our correct behavior.